### ISSUE TITLE
PubSub test failure

### ISSUE THREAD

**Issue by jskeet** [GoogleCloudPlatform/google-cloud-dotnet#1211]
I've just received this test failure running the integration tests:

```text
BulkMessaging test
50 messages; of size [9900000, 9990000]; max in-flight: 20, initialNacks: 0
Topic, Subscription, SimplePublisher and SimpleSubscriber all created
Sent: 4; Recv: 0
Sent: 9; Recv: 0
Sent: 15; Recv: 0
Sent: 20; Recv: 0
Sent: 20; Recv: 0
Sent: 20; Recv: 0
Sent: 20; Recv: 0
Sent: 20; Recv: 0
Sent: 20; Recv: 0
Sent: 20; Recv: 0
Sent: 20; Recv: 0
Sent: 20; Recv: 0
Sent: 20; Recv: 0
Sent: 20; Recv: 0
Sent: 20; Recv: 0
Deadlock detected. Cancelling test
[xUnit.net 00:04:14.8115138]     Google.Cloud.PubSub.V1.IntegrationTests.SimplePubSubTest.MaximumSizedMessages [FAIL]
[xUnit.net 00:04:14.8135260]       Test cancelled by watchdog
[xUnit.net 00:04:14.8137153]       Expected: True
[xUnit.net 00:04:14.8137733]       Actual:   False
[xUnit.net 00:04:14.8152352]       Stack Trace:
[xUnit.net 00:04:14.8191241]         C:\Users\jon\Test\Projects\google-cloud-dotnet\apis\Google.Cloud.PubSub.V1\Google.Cloud.PubSub.V1.IntegrationTests\SimplePubSubTest.cs(130,0): at Google.Cloud.PubSub.V1.IntegrationTests.SimplePubSubTest.<RunBulkMessaging>d__2.MoveNext()
[xUnit.net 00:04:14.8193517]         --- End of stack trace from previous location where exception was thrown ---
[xUnit.net 00:04:14.8194257]            at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
[xUnit.net 00:04:14.8194797]            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
[xUnit.net 00:04:14.8195511]         C:\Users\jon\Test\Projects\google-cloud-dotnet\apis\Google.Cloud.PubSub.V1\Google.Cloud.PubSub.V1.IntegrationTests\SimplePubSubTest.cs(178,0): at Google.Cloud.PubSub.V1.IntegrationTests.SimplePubSubTest.<MaximumSizedMessages>d__5.MoveNext()
[xUnit.net 00:04:14.8196387]         --- End of stack trace from previous location where exception was thrown ---
[xUnit.net 00:04:14.8196915]            at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
[xUnit.net 00:04:14.8197462]            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
[xUnit.net 00:04:14.8197856]         --- End of stack trace from previous location where exception was thrown ---
[xUnit.net 00:04:14.8198400]            at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
[xUnit.net 00:04:14.8198954]            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
[xUnit.net 00:04:14.8199669]         --- End of stack trace from previous location where exception was thrown ---
[xUnit.net 00:04:14.8200183]            at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
[xUnit.net 00:04:14.8200606]            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Failed   Google.Cloud.PubSub.V1.IntegrationTests.SimplePubSubTest.MaximumSizedMessages
Error Message:
 Test cancelled by watchdog
Expected: True
Actual:   False
Stack Trace:
   at Google.Cloud.PubSub.V1.IntegrationTests.SimplePubSubTest.<RunBulkMessaging>d__2.MoveNext() in C:\Users\jon\Test\Projects\google-cloud-dotnet\apis\Google.Cloud.PubSub.V1\Google.Cloud.PubSub.V1.IntegrationTests\SimplePubSubTest.cs:line 130
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Google.Cloud.PubSub.V1.IntegrationTests.SimplePubSubTest.<MaximumSizedMessages>d__5.MoveNext() in C:\Users\jon\Test\Projects\google-cloud-dotnet\apis\Google.Cloud.PubSub.V1\Google.Cloud.PubSub.V1.IntegrationTests\SimplePubSubTest.cs:line 178
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

The other tests passed.

**Comment by chrisdunelm** [comment]
What machine were you running this on? A desktop/laptop or a GCE machine?

**Comment by jskeet** [comment]
Laptop from home. Not sure why the first would fail but the later ones work.

**Comment by chrisdunelm** [comment]
The test that failed is sending must larger messages than the other tests. I'm not sure why this would fail though.
Did this happen just once? Or did you try multiple times?

Looking at the test console output, it stopped when it had sent 20 messages, which means that almost certainly zero messages got sent successfully to the server as 20 is the max allowed in-flight for that test.
It's possible that setting the publisher max message size to unlimited in #1212 may help.

**Comment by jskeet** [comment]
Only tried once. Happy to run the tests more times, and try the fix in #1212... leave that with me.

**Comment by jskeet** [comment]
I've now tried it 4 times, and it's failed every time - and on the last attempt, I was using the attempted fix, to no avail :(

**Comment by chrisdunelm** [comment]
What upload bandwidth do you have? It could be that it takes too long to upload a single message, so it's timing out.
If you increase the "10" in [SimplePubSubTest.cs line 104](https://github.com/GoogleCloudPlatform/google-cloud-dotnet/blob/master/apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.IntegrationTests/SimplePubSubTest.cs#L104) to, say, 100; does that change anything?

**Comment by jskeet** [comment]
I have 49Mbps down, and 17Mbps up, which I would expect to be enough. Testing the change now...

**Comment by jskeet** [comment]
Still failed after waiting for 100 iterations :(

**Comment by jskeet** [comment]
This is now fixed by #1212.
